build.py

```diff
@@ -23,4 +31,177 @@ def decrypt():
 @task(_checkdir)
 def encrypt():
     '''Encrypt secrets.dec.tf file.'''
-    os.system("ansible-vault encrypt --output secrets.enc secrets.dec.tf && rm -rf secrets.dec.tf")
\ No newline at end of file
+    os.system("ansible-vault encrypt --output secrets.enc secrets.dec.tf && rm -rf secrets.dec.tf")
+
+@task(_checkdir)
+def layer_dependency(summary='False',quiet='False'):
+    """Check layer dependency from the current layers"""
+    try:
+        remote_states = _layer_dependency(path.get_working_path())
+    except Exception as e:
+        print('Error getting deps:',e)
+        sys.exit(0)
+
+    if summary.lower() == 'true':
+        deps = {'this': []}
+        for rs in remote_states:
+            deps['this'].append(remote_states[rs]['key'])
+        remote_states = deps
+
+    try:
+        if quiet == 'False': print("Note layer dependency is calculated using remote states.\nNevertheless, other sort of dependencies could exist without this kind of resources,\ne.g. if you relay on some resource created in a different layer and not referenced here.")
+        if len(remote_states) > 0:
+            print(json.dumps(remote_states, indent=1))
+        else:
+            if quiet == 'False': print('No dependency detected.')
+    except Exception as e:
+        print('Error processing deps:',e)
+        sys.exit(0)
+
+def _layer_dependency(directory):
+
+    config_tfvars = _load_tfvars()
+    config_locals = _load_locals()
+
+    config_files = ['config.tf']
+    config_file_contents = []
+
+    # #########################################
+    # get the config file contents
+    # #########################################
+    try:
+      for config_file in config_files:
+          with open(f"{directory}/{config_file}", 'r') as f:
+              config_file_contents.append(hcl2.load(f))
+    except Exception as e:
+        raise Exception(f"Can not open config files: {e}")
+
+    remote_states = {}
+
+    if len(config_file_contents)>0:
+        # #########################################
+        # Where data is being used
+        # #########################################
+        remote_state_usage_regex = 'data\.terraform_remote_state\.([a-z_\-0-9]+)'
+        files = glob.glob(f"{directory}/*tf")
+        remote_state_lines = {}
+        for this_file in files:
+            with open(this_file, 'r') as f:
+                this_file_data = f.readlines()
+                for line in this_file_data:
+                    if match_object := re.search(remote_state_usage_regex, line):
+                        if not match_object.group(1) in remote_state_lines.keys():
+                            remote_state_lines[match_object.group(1)] = [this_file]
+                        else:
+                            if not this_file in remote_state_lines[match_object.group(1)]:
+                                remote_state_lines[match_object.group(1)].append(this_file)
```

Comment:
A `defaultdict(list)` would be the ideal structure to use in here, simply appending blindly like in the last line for every match would take care of this logic